### DISCUSSION TITLE
Not all modules in __init__, intentional?

### DISCUSSION THREAD

**Discussion by Adamasterr** [gijzelaerr/python-snap7#384] [Q&A] [answered]
Hello,

I am trying to use Partner but I see that it is not listed in `__init__.py`, however, I can see `partner.py` in the snap7 folder.
Same thing I noticed for `exceptions.py`.
Can you please help me understand why? Is it a bug? Is it not yet ready? Am I not allowed to use it?

I am just a beginner in using this library so I might have missed something.

Thank you :)

**Comment by swamper123** [comment, marked as answer]
The `exception.py` is a leftover of older versions.
Since #272 many parts have been replaced and I see that it isn't used anymore anywhere (wondering that no linter hit on that ...). So that file is irrelevant.
I don't know why `partner.py` is missing, but I guess most of the users of that library use it for client-server usecases and nobody cared so far. Tbh, I am not sure if that part was even tested well on real machines so far.

**Reply by Adamasterr** [reply]
Thank you, that makes sense.
